File: backend/src/main/java/vn/nguongocso/event/enums/ChainEventType.java
package vn.nguongocso.event.enums;

/**
 * Danh sách các loại sự kiện trong vòng đời chuỗi cung ứng.
 * 
 * @author Triệu Văn Đại
 */

public enum ChainEventType {
    // Sự kiện thu hoạch
    HARVEST, // Thu hoạch

    // Sự kiện đóng gói - Sản phẩm được đóng gói và dán nhãn.
    PACKAGING, // Đóng gói

    // Sự kiện vận chuyển - Sản phẩm được di chuyển giữa các địa điểm.
    TRANSPORT, // Vận chuyển

    // Sự kiện thu mua - Sản phẩm được mua hoặc tiếp nhận.
    PROCUREMENT, // Thu mua

    // Sự kiện sửa lỗi - Điều chỉnh hoặc sửa dữ liệu sự kiện trước đó.
    CORRECTION, // Sửa lỗi

    // Sự kiện nhập kho và đối chiếu số lượng - Doanh nghiệp thu mua ghi nhận số lượng thực nhận.
    WAREHOUSE_RECEIPT, // Nhập kho

    // Sự kiện theo dõi điều kiện bảo quản khi vận chuyển.
    STORAGE_CONDITION // Theo dõi bảo quản
}

File: frontend/src/enums/chainEventType.ts
// Dùng object thay cho enum để tương thích với erasableSyntaxOnly
export const ChainEventType = {
  HARVEST: 'HARVEST',
  PACKAGING: 'PACKAGING',
  TRANSPORT: 'TRANSPORT',
  PROCUREMENT: 'PROCUREMENT',
  CORRECTION: 'CORRECTION',
  WAREHOUSE_RECEIPT: 'WAREHOUSE_RECEIPT',
  STORAGE_CONDITION: 'STORAGE_CONDITION',
} as const;

export type ChainEventType = (typeof ChainEventType)[keyof typeof ChainEventType];

// Nhãn hiển thị tiếng Việt
export const ChainEventTypeLabel: Record<ChainEventType, string> = {
  [ChainEventType.HARVEST]: 'Thu hoạch',
  [ChainEventType.PACKAGING]: 'Đóng gói',
  [ChainEventType.TRANSPORT]: 'Vận chuyển',
  [ChainEventType.PROCUREMENT]: 'Thu mua',
  [ChainEventType.CORRECTION]: 'Đính chính',
  [ChainEventType.WAREHOUSE_RECEIPT]: 'Nhập kho',
  [ChainEventType.STORAGE_CONDITION]: 'Điều kiện bảo quản',
};

// English display labels for event types
export const ChainEventTypeEnLabel: Record<ChainEventType, string> = {
  [ChainEventType.HARVEST]: 'Harvesting',
  [ChainEventType.PACKAGING]: 'Packaging',
  [ChainEventType.TRANSPORT]: 'Transport',
  [ChainEventType.PROCUREMENT]: 'Procurement',
  [ChainEventType.CORRECTION]: 'Correction',
  [ChainEventType.WAREHOUSE_RECEIPT]: 'Warehouse Receipt',
  [ChainEventType.STORAGE_CONDITION]: 'Storage Condition',
};
